screens/Home/HomeScreen.tsx

```diff
@@ -1,78 +1,89 @@
+import { Product } from "../../models/products.model";
+
 
 const HomeScreen = () => {
+  const { data, error, isLoading, isFetching, isSuccess } = useProductsQuery()
+
   const navigation = useNavigation();
 
-  return (
-    <View style={{ flex: 1, backgroundColor: "#dcb688" }}>
+  const data2 = data?.slice(3, -5)
+  const data3 = data2?.splice(2, 8)
+
+  console.log(data)
+  const renderCategories = (data: Product[]) => {
+    const categories = data.map((product, index) => {
+      return (
+        <Category
+          key={index}
+          index={index}
+          image={product.image}
+          category={product.category}
+        />
+      )
+    });
+    return categories;
+  }
+
+  return (<SafeAreaView style={{ flex: 1, backgroundColor: "#dcb688" }}>
+
+    {isLoading && <Text>...Loading </Text>}
+    {isFetching && <Text>...Fetching </Text>}
+    {error && <Text>...Something went wrong </Text>}
+    {isSuccess && (<ScrollView showsVerticalScrollIndicator={false} style={{ backgroundColor: "#dcb688", padding: 10 }}>
       <Text style={styles.title}>Our Products</Text>
-      <View style={{ alignItems: "center" }}>
-        <View style={styles.firstContainer}>
-          <View
-            style={{
-              flex: 1,
-              flexDirection: "row",
-              justifyContent: "space-between",
-            }}
-          >
-            <TouchableOpacity style={styles.view1}
-              onPress={() => navigation.navigate("ProductDetailsScreen" as never)}></TouchableOpacity>
-            <TouchableOpacity style={styles.view1}
-              onPress={() => navigation.navigate("ProductDetailsScreen" as never)}
-            ></TouchableOpacity>
-          </View>
-          <View
-            style={{
-              flex: 1,
-              flexDirection: "row",
-              justifyContent: "space-between",
-              marginTop: 40,
-            }}
-          >
-            <TouchableOpacity style={styles.view1}></TouchableOpacity>
-            <TouchableOpacity style={styles.view1}></TouchableOpacity>
+
+      <View >
+        <View style={{ alignItems: "center" }}>
+          < View style={styles.firstContainer}>
+            {data2?.map((product, index) => {
+              return (
+                <View key={index}
+                >
+                  <TouchableOpacity style={styles.view1}
+                    onPress={() => navigation.navigate("ProductDetailsScreen" as never, { id: product.id } as never)}
+                  >
+                    <Image
+                      resizeMode="cover"
+                      style={{
+                        width: 125,
+                        height: 135,
+                        borderRadius: 15,
+
+                        aspectRatio: 1,
+                      }}
+                      source={{ uri: `${product.image}` }}
+                    />
+                  </TouchableOpacity>
+                </View>
+              )
+            })}
```

Comment:
You could show this only when we have data from the API like `categories && <the rest of the code>`